ci/taos/checker-pr-gateway.sh

```diff
@@ -135,34 +135,70 @@ $checkout_branch
-# Prog1 and Prog2 woudl be started in the background.
-# --------------------------- Run a checker: format group ---------------------------------------
-cd $dir_ci
-echo -e "[DEBUG] dir_commit is $dir_commit. This folder is created." | tee    $log_file_format
-echo -e "[DEBUG] current path: $(pwd)."                              | tee -a $log_file_format
-# Save a log file to debug a format cheker
-log_file_format="${dir_ci}/${dir_commit}/checker-pr-format.log"
-echo -e "[DEBUG] ./checker-pr-format-async.sh $1 $2 $3 $4 $5 $6 "    | tee -a $log_file_format
-echo -e "[DEBUG] Starting a format checker...            "           | tee -a $log_file_format
-# Run format checker
-pushd ./taos/
-./checker-pr-format-async.sh $1 $2 $3 $4 $5 $6                       | tee -a $log_file_format &
-popd
-echo -e "[DEBUG] Running..."                                         | tee -a $log_file_format
-echo -e "[DEBUG] Completed."                                         | tee -a $log_file_format
-
-# --------------------------- Run a checker: audit group ----------------------------------------
-cd $dir_ci
-echo -e "[DEBUG] dir_commit is $dir_commit. This folder is created." | tee    $log_file_audit
-echo -e "[DEBUG] current path: $(pwd)."                              | tee -a $log_file_audit
-# Save a log file to debug a audit checker
-log_file_audit="${dir_ci}/${dir_commit}/checker-pr-audit.log"
-echo -e "[DEBUG] ./checker-pr-audit-async.sh $1 $2 $3 $4 $5 $6 "     | tee -a $log_file_audit
-echo -e "[DEBUG] Starting a audit checker...             "           | tee -a $log_file_audit
-# Run audit checker
-pushd ./taos/
-./checker-pr-audit-async.sh $1 $2 $3 $4 $5 $6                        | tee -a $log_file_audit &
-popd
-echo -e "[DEBUG] Running..."                                         | tee -a $log_file_audit
-echo -e "[DEBUG] Completed."                                         | tee -a $log_file_audit
+
+# @brief checker runner to run the checkers based on the dependency policy
+# Run the first group, then depending on the dependency policy, run the remaining groups
+# @param arguments received by the function
+#  arg1: list of checkers (like format, audit, etc)
+#  arg2: list of checker's name
+#  arg3: list of checker's log file
+#  arg@: all remaining arguments are to be passed to the checkers (common for all the checkers)
+function run_all_checkers(){
+  local -n cmd_list=$1
+  local -n name_list=$2
+  local -n logfile_list=$3
+  local checker_args=${@:4}
+
+  local num_checkers=${#cmd_list[@]}
+  for (( i=0; i<$num_checkers; i++ )); do
+    local logfile=${logfile_list[$i]}
+
+    cd $dir_ci
+    echo -e "[DEBUG] dir_commit is $dir_commit. This folder is created." | tee    $logfile
+    echo -e "[DEBUG] current path: $(pwd)."                              | tee -a $logfile
+    # Save a log file to debug the cheker
+    echo -e "[DEBUG] ${cmd_list[$i]} $checker_args "             | tee -a $logfile
```

Comment:
Please keep the same location of "| tee -a ....." ASAP.

Reply:
updated